evm_trace/geth.py

```diff
@@ -3,18 +3,18 @@
 
 from eth_utils import to_int
 from ethpm_types import HexBytes
+from pydantic import Field, RootModel, field_validator
 
-from evm_trace._pydantic_compat import Field, validator
 from evm_trace.base import BaseModel, CallTreeNode
 from evm_trace.enums import CALL_OPCODES, CallType
 from evm_trace.utils import to_address
 
 
-class TraceMemory(BaseModel):
-    __root__: List[HexBytes] = []
+class TraceMemory(RootModel[List[HexBytes]]):
+    root: List[HexBytes] = []
```

Comment:
I still defined `root` here because that was the only way to make it default to empty list when it got used in other models.